From edb68aa0f57095bddbc7449b467fd1fcfa5e90f7 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:10:26 -0400
Subject: [powerpc] powernv: Add invalid OPAL call

Message-id: <1410545655-205645-398-git-send-email-dzickus@redhat.com>
Patchwork-id: 94396
O-Subject: [RHEL7 PATCH 397/626] powerpc/powernv: Add invalid OPAL call
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit e28b05e7ae8ba09e030ffe891ba154df5791cb76
Author: Joel Stanley <joel@jms.id.au>
Date:   Tue Apr 1 14:28:20 2014 +1030

    powerpc/powernv: Add invalid OPAL call

    This call will not be understood by OPAL, and cause it to add an error
    to it's log. Among other things, this is useful for testing the
    behaviour of the log as it fills up.

    Signed-off-by: Joel Stanley <joel@jms.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/opal.h b/arch/powerpc/include/asm/opal.h
index 05f9455..6bd3b18 100644
--- a/arch/powerpc/include/asm/opal.h
+++ b/arch/powerpc/include/asm/opal.h
@@ -87,6 +87,7 @@ extern int opal_enter_rtas(struct rtas_args *args,
 #define OPAL_ASYNC_COMPLETION -15
 
 /* API Tokens (in r0) */
+#define OPAL_INVALID_CALL   -1
 #define OPAL_CONSOLE_WRITE   1
 #define OPAL_CONSOLE_READ   2
 #define OPAL_RTC_READ    3
@@ -734,6 +735,7 @@ extern struct kobject *opal_kobj;
 extern struct device_node *opal_node;
 
 /* API functions */
+int64_t opal_invalid_call(void);
 int64_t opal_console_write(int64_t term_number, __be64 *length,
       const uint8_t *buffer);
 int64_t opal_console_read(int64_t term_number, __be64 *length,
diff --git a/arch/powerpc/platforms/powernv/opal-wrappers.S b/arch/powerpc/platforms/powernv/opal-wrappers.S
index bb90f9a..f531ffe 100644
--- a/arch/powerpc/platforms/powernv/opal-wrappers.S
+++ b/arch/powerpc/platforms/powernv/opal-wrappers.S
@@ -61,6 +61,7 @@ _STATIC(opal_return)
  mtcr r4;
  rfid
 
+OPAL_CALL(opal_invalid_call,   OPAL_INVALID_CALL);
 OPAL_CALL(opal_console_write,   OPAL_CONSOLE_WRITE);
 OPAL_CALL(opal_console_read,   OPAL_CONSOLE_READ);
 OPAL_CALL(opal_console_write_buffer_space, OPAL_CONSOLE_WRITE_BUFFER_SPACE);
diff --git a/arch/powerpc/platforms/powernv/opal.c b/arch/powerpc/platforms/powernv/opal.c
index 99e9c28..49d2f00 100644
--- a/arch/powerpc/platforms/powernv/opal.c
+++ b/arch/powerpc/platforms/powernv/opal.c
@@ -635,3 +635,6 @@ void opal_shutdown(void)
    mdelay(10);
  }
 }
+
+/* Export this so that test modules can use it */
+EXPORT_SYMBOL_GPL(opal_invalid_call);
-- 
1.7.1